rfc/0008-upgrade-statement.md

```diff
@@ -0,0 +1,103 @@
+|              |                                  |
+| :----------- | :------------------------------- |
+| Feature Name | Upgrade Statement                |
+| Start Date   | 05/23/2022                       |
+| Category     | updates and versioning           |
+| RFC PR       | https://github.com/kubewarden/rfc/pull/6  |
+| State        | ~**ACCEPTED**~                   |
+
+
+# Summary
+[summary]: #summary
+
+This RFC proposes how the upgrade of the Kuberwarden stack will work and how to do it.
+
+# Motivation
+[motivation]: #motivation
+
+Kubewarden is reaching the v1.0 milestone and we need to define a upgrade process
+to users and for the development team to follow. Therefore, users will are be able
+to plan when and how to upgrade their Kuberwarden stack. And the Kuberwarden developers
+will be able to define when and how ship new features and fixes.
+
+## Examples / User Stories
+[examples]: #examples
+
+As Kubewarden user, I want to know, among all Kubewarden components, what are the
+compatible versions.
+
+As Kubewarden user, I want to know the upgrade path of the Kuberwarden components.
+
+As Kubewarden developer, I want to know which components I need to upgrade in order
+to release a feature of a bug fix.
+
+# Detailed design
+[design]: #detailed-design
+
+This document describes how Kubewarden versioning works and how users can upgrade
+their stack. The Kubewarden project uses [Semantic versioning](https://semver.org/)
+to define the version of all its components. In other words, the version follows
+the `MAJOR.MINOR.PATCH` pattern.  The supported versions are the two latest releases.
+This means that if the latest version released of the Kubewarden components is
+`1.2.0`. Both `1.2.0` and `1.1.0` version are under active maintenance.
+
+The Kubewarden components that follow the rules described here are:
+
+- `kubewarden-controller`
+- `policy-server`
+- `kwctl`
+- Policies
+- SDKs
+
+`kubewarden-controller`, `policy-server` and `kwctl` share the same `MAJOR`
+and `MINOR` version. The `PATCH` version can increase independently though.
+
+## Version compatibility among components
+
+`kubewarden-controller`, `policy-server` and `kwctl` should run the same `MAJOR`/`MINOR`.
+Therefore, if the `kubewarden-controller` version running is `1.1.x`, the
+`policy-server` and `kwctl` version in use should be `1.1.x` as well.
+
+Considering that `kwctl` and `policy-server` should run policies from older versions.
+Policies and the SDKs used to write them are independent from the other components.
+Their version does not need to be synchronized with the remain of the components.
+
```

Comment:
There might be policies that needs a new version of the policy server. Imagine we add a new host capability, then we create a policy that uses it. It will work just with the version of the policy server that contains this change, and it will not work with previous version. E.g verify-image-signatures policy won't work with old versions of policy-server.

I think it should be fine if we just mention in the policy which version of Kubewarden you need. What do you think?

Reply:
> I think it should be fine if we just mention in the policy which version of Kubewarden you need. What do you think?

Yes, this is something we can mention inside of the policy. Maybe, but this is not related to this RFC, we could add a new entry inside of the policy metadata to specify which version of the host capabilities is needed. Then, at load time, policy-server/kwctl can fail if it cannot satisfy this requirement

Reply:
As we removed the policies and SDKs as components from this RFC. I'll remove this piece of text.